spec/02-integration/14-tracing/03-tracer-pdk_spec.lua

```diff
@@ -0,0 +1,81 @@
+local helpers = require "spec.helpers"
+local cjson   = require "cjson"
+local join    = require "pl.stringx".join
+
+
+local TCP_PORT = 35001
+local tcp_trace_plugin_name = "tcp-trace-exporter"
+
+
+for _, strategy in helpers.each_strategy() do
+  local proxy_client
+
+  describe("tracer pdk spec #" .. strategy, function()
+
+    local function setup_instrumentations(types, custom_spans, sampling_rate)
+      local bp, _ = assert(helpers.get_db_utils(strategy, {
+        "services",
+        "routes",
+        "plugins",
+      }, { tcp_trace_plugin_name }))
+
+      local http_srv = assert(bp.services:insert {
+        name = "mock-service",
+        host = helpers.mock_upstream_host,
+        port = helpers.mock_upstream_port,
+      })
+
+      bp.routes:insert({ service = http_srv,
+                         protocols = { "http" },
+                         paths = { "/" }})
+
+      bp.plugins:insert({
+        name = tcp_trace_plugin_name,
+        config = {
+          host = "127.0.0.1",
+          port = TCP_PORT,
+          custom_spans = custom_spans or false,
+        }
+      })
+
+      assert(helpers.start_kong {
+        database = strategy,
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        plugins = "tcp-trace-exporter",
+        tracing_instrumentations = types,
+        tracing_sampling_rate = sampling_rate or 1,
+      })
+
+      proxy_client = helpers.proxy_client()
+    end
+
+    describe("sampling rate", function ()
+      local instrumentations = { "request", "router", "balancer" }
```

Comment:
Just a suggestion. Maybe we could randomize the instrumentations being enabled.

Reply:
we could use "all", here I just picked the instrumentations that are more stable (i.e. don't depend on db queries, dns, plugin phases, etc), for this test in particular we only need "more than one span" to be created with a parent/child relationship so we can ensure sampling is applied correctly